numba/cpython/setobj.py

```diff
@@ -952,6 +950,7 @@ def _resize(self, payload, nentries, errmsg):
             entry = loop.entry
             self._add_key(payload, entry.key, entry.hash,
                           do_resize=False)
+            context.nrt.decref(builder, self._ty.dtype, entry.key)
 
         self._free_payload(old_payload.ptr)
```

Comment:
This seems to deallocate the pointer after realloc which doesn't seem valid to me. This whole code section isn't necessary since realloc copies the original data: https://en.cppreference.com/w/c/memory/realloc